### ISSUE TITLE
Feature Request: Comprehensive Synthetic Tabular Data Generation Models

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#827]
## Overview

This is a comprehensive feature request to add state-of-the-art synthetic tabular data generation models to AiDotNet. These models are essential for data augmentation, privacy-preserving data sharing, and training ML models on small datasets.

### 1. CTGAN (Conditional Tabular GAN) - CRITICAL
- **Paper**: "Modeling Tabular Data using Conditional GAN" (NeurIPS 2019) - https://arxiv.org/abs/1907.00503
- **Why Needed**: Industry standard for mixed-type tabular data
- Mode-specific normalization for multi-modal distributions
- Conditional vector for handling categorical columns
- PacGAN-style training for diversity

### 2. TVAE (Tabular VAE) - HIGH
- Same paper as CTGAN
- Faster training than GAN, more stable for smaller datasets
- Evidence Lower Bound (ELBO) loss

### 3. CopulaGAN - HIGH
- **Paper**: "Synthesizing Tabular Data using Copulas" (2020)
- Better preservation of statistical dependencies
- Gaussian copula transformation

### 4. TableGAN - MEDIUM
- **Paper**: "TableGAN: Synthesizing Tabular Data with GAN" (2018)
- Privacy-focused generation with label column optimization

### 5. TabDDPM (Diffusion for Tabular) - HIGH
- **Paper**: "TabDDPM: Modelling Tabular Data with Diffusion Models" (ICML 2023)
- Outperforms GANs on many benchmarks
- Multinomial diffusion for categoricals
- Gaussian diffusion for numericals

### 6. CTAB-GAN / CTAB-GAN+ - HIGH
- **Paper**: "CTAB-GAN: Effective Table Data Synthesizing" (2021)
- Mixed-type encoder with mode-specific layers
- Conditional generator with auxiliary classifier

### 7. TabTransformer-based Generation - MEDIUM
- **Paper**: "TabTransformer: Tabular Data Modeling Using Contextual Embeddings" (2020)
- Contextual embeddings for categoricals
- Column-wise attention

### 8. REaLTabFormer - MEDIUM
- **Paper**: "REaLTabFormer: Generating Realistic Relational and Tabular Data using Transformers" (2023)
- GPT-2 architecture for sequential generation
- Parent-child relationship handling

### 9. GOGGLE - MEDIUM
- **Paper**: "GOGGLE: Generative Modelling for Tabular Data by Learning Relational Structure" (ICLR 2023)
- Graph neural network encoder
- Feature dependency modeling

### 10. TabSyn - HIGH
- **Paper**: "Bridging the Gap: Plug-and-Play Adaptation for Improved Tabular Generation" (NeurIPS 2023)
- VAE latent diffusion
- Latest SOTA combining VAE and diffusion

### 11. TabFlow - HIGH
- Flow matching for tabular data (2024)
- Optimal transport paths
- Score-based generation

### 12. AutoDiff-Tab - MEDIUM
- Automated Diffusion Models for Tabular Data (2024)
- Automated architecture search
- Noise schedule optimization

### 13. TabLLM-Gen - MEDIUM
- LLM-based Tabular Data Generation (2024)
- LLM instruction-following for schema understanding
- Few-shot generation

### 14. FinDiff - SPECIALIZED
- Diffusion Models for Financial Tabular Data (2024)
- Temporal correlation preservation
- Volatility modeling

### 15. MedSynth - SPECIALIZED
- Privacy-Preserving Medical Tabular Synthesis (2024)
- Differential privacy guarantees
- Clinical validity constraints

### 16. DP-CTGAN - HIGH
- Differentially Private Tabular Data Generation
- Epsilon-differential privacy
- Gradient clipping and noise injection

### 17. PATE-GAN - MEDIUM
- **Paper**: "PATE-GAN: Generating Synthetic Data with Differential Privacy Guarantees" (ICLR 2019)
- Teacher-student framework
- Private aggregation

### 18. AIM (Adaptive Iterative Mechanism) - MEDIUM
- Marginal-based synthesis
- Workload-aware privacy

### 19. OCT-GAN (One-Class Tabular GAN) - LOW
- For imbalanced data augmentation
- Minority class oversampling

### 20. TimeGAN - MEDIUM
- **Paper**: "Time-series Generative Adversarial Networks" (NeurIPS 2019)
- Temporal embedding for sequential tabular data
- Autoregressive generation

### 21. MisGAN (Missing Data GAN) - LOW
- **Paper**: "MisGAN: Learning from Incomplete Data with GANs" (ICLR 2019)
- Missing mask modeling
- MCAR/MAR/MNAR handling

### 22. Causal-GAN - LOW
- Causal graph structure
- Intervention simulation
- Counterfactual generation

### 23. Bayesian Network Synthesis - LOW
- DAG structure learning
- Conditional probability tables

### 24. Copula-Based Synthesis (Non-Deep) - LOW
- Various copula families (Gaussian, Clayton, Frank, Gumbel)
- Marginal fitting

### 25. SMOTE variants for mixed types - LOW
- SMOTE-NC for categorical + numerical

## Proposed API Design

```csharp
public interface ISyntheticTabularGenerator<T>
{
    Task FitAsync(TabularDataset<T> data, GeneratorOptions options, CancellationToken ct);
    TabularDataset<T> Generate(int numSamples);
    double[,] GenerateArray(int numSamples);
}

public class CTGANGenerator<T> : ISyntheticTabularGenerator<T> { }
public class TVAEGenerator<T> : ISyntheticTabularGenerator<T> { }
public class TabDDPMGenerator<T> : ISyntheticTabularGenerator<T> { }
```

## Implementation Priority

| Priority | Model | Complexity |
|----------|-------|------------|
| P0 | CTGAN | Medium |
| P0 | TVAE | Medium |
| P0 | TabDDPM | High |
| P1 | CopulaGAN | Medium |
| P1 | CTAB-GAN+ | High |
| P1 | TabSyn | High |
| P2 | REaLTabFormer | High |
| P2 | DP-CTGAN | Medium |
| P2 | TimeGAN | Medium |

## Related Work

Current GenerativeAdversarialNetwork<T> lacks:
- Mode-specific normalization for tabular data
- Conditional generation capabilities
- Mixed-type (categorical + numerical) handling
- Privacy-preserving mechanisms

**Requested by**: TDAB Platform Team
**Use Case**: SyntheticBoost feature for data augmentation
**Priority**: HIGH
